File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/app/d/CiAppLisSheetOrDO.java
package iih.ci.ord.app.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.ord.app.d.desc.CiAppLisSheetOrDODesc;
import java.math.BigDecimal;

/**
 * 检验申请单对应医嘱 DO数据 
 * 
 */
public class CiAppLisSheetOrDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_CIAPPLISSHEETOR= "Id_ciapplissheetor";
	public static final String ID_CIAPPLISSHEET= "Id_ciapplissheet";
	public static final String ID_ORLAB= "Id_orlab";
	public static final String ID_OR= "Id_or";
	public static final String AMT_APP= "Amt_app";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_ciapplissheetor() {
		return ((String) getAttrVal("Id_ciapplissheetor"));
	}	
	public void setId_ciapplissheetor(String Id_ciapplissheetor) {
		setAttrVal("Id_ciapplissheetor", Id_ciapplissheetor);
	}
	public String getId_ciapplissheet() {
		return ((String) getAttrVal("Id_ciapplissheet"));
	}	
	public void setId_ciapplissheet(String Id_ciapplissheet) {
		setAttrVal("Id_ciapplissheet", Id_ciapplissheet);
	}
	public String getId_orlab() {
		return ((String) getAttrVal("Id_orlab"));
	}	
	public void setId_orlab(String Id_orlab) {
		setAttrVal("Id_orlab", Id_orlab);
	}
	public String getId_or() {
		return ((String) getAttrVal("Id_or"));
	}	
	public void setId_or(String Id_or) {
		setAttrVal("Id_or", Id_or);
	}
	public FDouble getAmt_app() {
		return ((FDouble) getAttrVal("Amt_app"));
	}	
	public void setAmt_app(FDouble Amt_app) {
		setAttrVal("Amt_app", Amt_app);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_ciapplissheetor";
	}
	
	@Override
	public String getTableName() {	  
		return "ci_app_lis_or";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(CiAppLisSheetOrDODesc.class);
	}
	
}

File: client/iih_stub/iih.ci_stub/iih.ci.ord/app/d/CiAppLisSheetOrDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;

namespace iih.ci.ord.app.d
{
    /// <summary>
    /// CiAppLisSheetOrDO 的摘要说明。
    /// </summary>
    public class CiAppLisSheetOrDO : BaseDO {

		public const string TABLE_NAME = "ci_app_lis_or";
		public const string TABLE_ALIAS_NAME = "a1";

        public CiAppLisSheetOrDO() {
        }
		public string Id_ciapplissheetor {
            get { return getAttrVal<string>("Id_ciapplissheetor",null); }
            set { setAttrVal<string>("Id_ciapplissheetor", value); }
        }
		public string Id_ciapplissheet {
            get { return getAttrVal<string>("Id_ciapplissheet",null); }
            set { setAttrVal<string>("Id_ciapplissheet", value); }
        }
		public string Id_orlab {
            get { return getAttrVal<string>("Id_orlab",null); }
            set { setAttrVal<string>("Id_orlab", value); }
        }
		public string Id_or {
            get { return getAttrVal<string>("Id_or",null); }
            set { setAttrVal<string>("Id_or", value); }
        }
		public FDouble Amt_app {
            get { return getAttrVal<FDouble>("Amt_app",null); }
            set { setAttrVal<FDouble>("Amt_app", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "ci_app_lis_or";
        }
        
        /// <summary>
        /// 返回表别名
        /// </summary>
        /// <returns></returns>
        public override string getAliasTableName()
        {
            return "a1";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_ciapplissheetor";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.ord.app.d.CiAppLisSheetOrDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_ciapplissheetor", "Id_ciapplissheet", "Id_orlab", "Id_or", "Amt_app"};
        }
        
    }
}
